r1cs/src/gadget/binary.rs

```diff
@@ -0,0 +1,58 @@
+use crate::driver::CircuitDriver;
+use crate::{R1cs, Wire};
+use std::marker::PhantomData;
+
+#[derive(Clone)]
+pub struct BinaryAssignment<C: CircuitDriver>(Wire, PhantomData<C>);
+
+impl<C: CircuitDriver> BinaryAssignment<C> {
+    pub fn instance(cs: &mut R1cs<C>, bit: u8) -> Self {
+        if bit != 0 && bit != 1 {
+            panic!("Bit value should be passed, got {bit}");
+        }
+        let wire = cs.public_wire();
+        cs.x.push(C::Scalar::from(bit as u64));
+
+        Self(wire, PhantomData::default())
+    }
+
+    pub fn witness(cs: &mut R1cs<C>, bit: u8) -> Self {
+        if bit != 0 && bit != 1 {
```

Comment:
we also check whether this bit is binary so we don't have to constrain with panic

Reply:
Removed